### ISSUE TITLE
`.pyi` file not detected for single `.py` file in site-packages

### ISSUE THREAD

**Issue by brettcannon** [python/mypy#15817]
**Bug Report**

A `.pyi` file next to a top-level `.py` file in `site-packages` is not being picked up.

P.S.: Docs are also inconsistent on naming of stubs-only packages. https://mypy.readthedocs.io/en/stable/installed_packages.html#installed-packages says `types-<name>` and then later says `<name>-stubs`.

**To Reproduce**

- Install `microvenv`
- Write a `microvenv.pyi` file in `site-packages` next to `microvenv.py`
- Run mypy with code importing `microvenv`

Example `microvenv.pyi`:
```python
import os

def create(env_dir: os.PathLike[str] | str = ...) -> None: ...
```

**Expected Behavior**

No error about missing type stubs or `py.typed`.

**Actual Behavior**

```
nox > mypy src/py_pip.py
src/py_pip.py:11: error: Skipping analyzing "microvenv": module is installed, but missing library stubs or py.typed marker  [import]
src/py_pip.py:11: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

I verified the `.pyi` file worked by copying to the same directory as the code being analyzed.

**Your Environment**

- Mypy version used: 1.4.1
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: Python 3.11.4

**Comment by JelleZijlstra** [comment]
This is correct behavior as specified by PEP 561. See python/typing#1333 for discussion of how to change the spec.
